packages/components/src/geolocation/useGeolocation.ts

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react';
+
+import { useGeolocationContext } from './GeolocationProvider';
+import type { GeolocationContextType } from './types';
+
+type Config = {
+  skip?: boolean;
+};
+
+export default function useGeolocation({
+  skip = false,
+}: Config = {}): GeolocationContextType {
+  const { resolve, called, coordinates, loading, error, ...delegated } =
+    useGeolocationContext();
+
+  useEffect(() => {
+    // Defer resolving under not skipped
+    // Don't resolve again if we have resolved before and found a geolocation
+    // Don't resolve if the previous resolve is ongoing
+    if (!skip && !called && coordinates === undefined && !loading) {
+      resolve();
+    }
+  }, [resolve, skip, called, coordinates, loading, error]);
```

Comment:
in dependency array there should be only the items that really matters. Otherwise it's called too often. At least the `error` is not even needed! I know the linter whines about the exhaustive deps, but the linter is wrong about it :)

Reply:
Removed the error as an unnecessary dependency but none of the others as they are all used.